src/lib/Form/EventListener/AddSiteAccessFieldBasedOnContentListener.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Form\EventListener;
+
+use eZ\Publish\API\Repository\Values\Content\Location;
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use EzSystems\EzPlatformAdminUi\Siteaccess\NonAdminSiteaccessResolver;
+use Symfony\Component\Form\ChoiceList\Loader\CallbackChoiceLoader;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\FormEvent;
+
+final class AddSiteAccessFieldBasedOnContentListener
+{
+    /** @var \EzSystems\EzPlatformAdminUi\Siteaccess\NonAdminSiteaccessResolver */
+    private $nonAdminSiteaccessResolver;
+
+    /** @var \eZ\Publish\Core\MVC\ConfigResolverInterface */
+    private $configResolver;
+
+    public function __construct(
+        NonAdminSiteaccessResolver $nonAdminSiteaccessResolver,
+        ConfigResolverInterface $configResolver
+    ) {
+        $this->nonAdminSiteaccessResolver = $nonAdminSiteaccessResolver;
+        $this->configResolver = $configResolver;
+    }
+
+    public function onPreSetData(FormEvent $event): void
+    {
+        /** @var \EzSystems\EzPlatformAdminUi\Form\Data\Content\CustomUrl\CustomUrlAddData $data */
+        $data = $event->getData();
+        $location = $data->getLocation();
+
+        $form = $event->getForm();
+
+        $form->add(
+            'root_location_id',
+            ChoiceType::class,
+            [
+                'required' => false,
+                'choice_loader' => new CallbackChoiceLoader($this->getCallableData($location)),
```

Comment:
Custom `ChoiceLoader` could be used here

Reply:
@mikadamczyk resolved in https://github.com/ezsystems/ezplatform-admin-ui/pull/1583/commits/0a598d578df2ae05c08ee8c40145daac086fe23a